module/VuFind/src/VuFind/RecordDriver/EDS.php

```diff
@@ -473,12 +473,29 @@ function ($data) {
      */
     public function getThumbnail($size = 'small')
     {
+        // Create a ranked list of sizes so we can use "best available" when appropriate.
+        // Note that "thumb" is a value used by EBSCO, not by VuFind; it is included so
+        // it can be matched up with requests for "small."
+        $sizes = ['thumb' => 0, 'small' => 1, 'medium' => 2, 'large' => 3];
+        $bestFit = -1;
+        $desiredFit = $sizes[$size] ?? count($sizes);
+        $closestMatch = false;
         foreach ($this->fields['ImageInfo'] ?? [] as $image) {
-            if ($size == ($image['Size'] ?? '')) {
-                return $image['Target'] ?? '';
+            $currentFit = $sizes[$image['Size'] ?? ''] ?? 0;
```

Comment:
Would coalescing to 'thumb' be clearer than double coalesing?

```suggestion
            $currentFit = $sizes[$image['Size'] ?? 'thumb'] ?? 0;
```